ui/imports/shared/status/StatusChatInput.qml

```diff
@@ -371,11 +371,11 @@ Rectangle {
             for (var i = 0; i < mentionsPos.length; i++) {
                 if (messageInputField.cursorPosition === mentionsPos[i].leftIndex) {
                     d.leftOfMentionIndex = i
-                    event.accepted = true
+                    event.accepted = (messageInputField.selectedText.length === 0);
```

Comment:
I think it's a good idea.
The only thing is that d.leftOfMentionIndex is assigned and used afterwards onRelease():
```
        if(d.rightOfMentionIndex !== -1) {
            //make sure to add an extra space between mention and text
            let mentionSeparator = event.key === Qt.Key_Space ? "" : "&nbsp;"
            messageInputField.insert(mentionsPos[d.rightOfMentionIndex].rightIndex, mentionSeparator + eventText)

            d.rightOfMentionIndex = -1
        }

        if(d.leftOfMentionIndex !== -1) {
            messageInputField.insert(mentionsPos[d.leftOfMentionIndex].leftIndex, eventText)

            d.leftOfMentionIndex = -1
        }
```


It will assume that mentionsPos[d.leftOfMentionIndex] is a valid object, but in our case when we select mention and hit text input key the mention will be deleted and mentionsPos[d.leftOfMentionIndex] is invalid.

I've done some tests where we check for selection in the if statements in order to avoid this and seems to be working fine:
```
if (messageInputField.cursorPosition === mentionsPos[i].leftIndex && messageInputField.selectedText.length === 0) {
...
} else if (messageInputField.cursorPosition === mentionsPos[i].rightIndex && messageInputField.selectedText.length === 0) {
...
```

What do you think of it?

Reply:
sure thing, updated!